libfoundation/src/system-library-ios.hpp

```diff
@@ -0,0 +1,193 @@
+/*                                                                     -*-c++-*-
+ Copyright (C) 2017 LiveCode Ltd.
+ 
+ This file is part of LiveCode.
+ 
+ LiveCode is free software; you can redistribute it and/or modify it under
+ the terms of the GNU General Public License v3 as published by the Free
+ Software Foundation.
+ 
+ LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+ WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ for more details.
+ 
+ You should have received a copy of the GNU General Public License
+ along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include "system-private.h"
+
+#include <foundation-auto.h>
+
+#if TARGET_IPHONE_SIMULATOR
+
+/* ================================================================
+ * iOS Simulator Handle Class
+ * ================================================================ */
+
+#include "system-library-mac.hpp"
+
+class __MCSLibraryHandleIOSSimulator: public __MCSLibraryHandleMac
+{
+};
+
+typedef class __MCSLibraryHandleIOSSimulator __MCSLibraryHandle;
+
+#else
+
+/* ================================================================
+ * iOS Device Handle Class
+ * ================================================================ */
+
+#include "system-library-static.hpp"
+#include "system-library-mac.hpp"
+
+class __MCSLibraryHandleIOSDevice
+{
+public:
+    __MCSLibraryHandleIOSDevice(void)
+        : m_type(kNone),
+          m_static()
+    {
+    }
+    
+    ~__MCSLibraryHandleIOSDevice(void)
+    {
+        switch(m_type)
+        {
+            case kNone:
+                break;
+            case kStatic:
+                m_static.~__MCSLibraryHandleStatic();
+                break;
+            case kDynamic:
+                m_dynamic.~__MCSLibraryHandleMac();
+                break;
+        }
+    }
+    
+    bool IsDefined(void) const
+    {
+        return m_type != kNone;
+    }
```

Comment:
It might be more "C++-ey" to define:

- `operator bool () const` instead of `IsDefined()`
- `operator ==(const __MCSLibraryHandleIOSDevice& p_other) const` instead of `IsEqualTo(...)`